segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollector.java

```diff
@@ -249,8 +249,8 @@ void addToGarbage(String chunkToDelete, long startTime, int attempts) {
         val chunksToDelete = new ArrayList<GarbageChunkInfo>();
         int count = 0;
         try {
-            // Block until you have at least one item.
-            GarbageChunkInfo info = garbageChunks.take();
+            // Wait until you have at least one item or timeout expires.
+            GarbageChunkInfo info = garbageChunks.poll(config.getGarbageCollectionDelay().toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
I don't suppose this will throw a TimeoutException, will it? I don't see how, as the code compiles and you don't rethrow it or otherwise catch it.

Reply:
And line 264 below is intentional, yes? We do not pass a timeout because we just want to know if something is immediately available or not. Is my understanding correct?

Reply:
(sorry but I cannot seem to be able to comment on line 267.) 

That is the wrong way to handle `InterruptedException` in Java. There are special semantics with that exception in that, if you catch it, the thread will become "un-interrupted", which may defeat the purpose of interrupting it. 

I strongly advise you use `Exceptions.handleInterruptible` for whatever call it is that may throw this. I suppose now it's the `poll(timeout)`.

Reply:
Double checked the documentation, it does not throw TimeoutException. 
It returns null.
> the head of this queue, or null if the specified waiting time elapses before an element with an expired delay becomes available

Reply:
yes line 264 is intentional. We do not want to wait for additional elements to be available.

Reply:
what happens if I don't catch InterruptedException ?

Reply:
done